File: Cosmos-IOS/Cosmostation/Cosmostation/Model/RizonSwapStatus.swift
//
//  RizonSwapStatus.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/07/30.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct RizonSwapStatus {
    var status: String?
    var _id: String?
    var from: String?
    var to: String?
    var amount: Double?
    var hdacTxId: String?
    var rizonTxId: String?
    var __v: UInt16?
    var hdacTx: HdacTx?
    
    init(_ dictionary: NSDictionary?) {
        self.status = dictionary?["status"] as? String
        self._id = dictionary?["_id"] as? String
        self.from = dictionary?["from"] as? String
        self.to = dictionary?["to"] as? String
        self.amount = dictionary?["amount"] as? Double
        self.hdacTxId = dictionary?["hdacTxId"] as? String
        self.rizonTxId = dictionary?["rizonTxId"] as? String
        self.__v = dictionary?["__v"] as? UInt16
        if let rawHdacTx = dictionary?["hdacTx"] as? NSDictionary {
            self.hdacTx = HdacTx.init(rawHdacTx)
        }
    }
}

public struct HdacTx {
    var txid: String?
    var confirmations: UInt64?
    var time: UInt64?
    var data: Array<String>?
    
    init(_ dictionary: NSDictionary?) {
        self.txid = dictionary?["txid"] as? String
        self.confirmations = dictionary?["confirmations"] as? UInt64
        self.time = dictionary?["time"] as? UInt64
        if let rawData = dictionary?["data"] as? Array<String> {
            self.data = rawData
        }
    }
    
}

File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/model/RizonSwapStatus.java
package wannabit.io.cosmostaion.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class RizonSwapStatus {
    @SerializedName("status")
    public String status;

    @SerializedName("_id")
    public String id;

    @SerializedName("from")
    public String from;

    @SerializedName("to")
    public String to;

    @SerializedName("amount")
    public Double amount;

    @SerializedName("hdacTxId")
    public String hdacTxId;

    @SerializedName("rizonTx")
    public Object rizonTx;

    @SerializedName("rizonTxId")
    public String rizonTxId;

    @SerializedName("__v")
    public int v;

    @SerializedName("hdacTx")
    public HdacTx hdacTx;

    public class HdacTx {
        @SerializedName("txid")
        public String txid;

        @SerializedName("confirmations")
        public int confirmations;

        @SerializedName("blockhash")
        public String blockhash;

        @SerializedName("blockheight")
        public long blockheight;

        @SerializedName("time")
        public long time;

        @SerializedName("valueOut")
        public double valueOut;

        @SerializedName("data")
        public ArrayList<String> data;

    }
}
